mlir-tensorrt/test/python/mlir_tensorrt_runtime/test_create_memref.py

```diff
@@ -496,6 +496,29 @@ def create_memref_from_dlpack(arr, module):
 # CHECK-NEXT: -- cupy.from_dlpack(): []
 
 
+def create_memref_dlpackview_from_dlpack(dl, module):
+    memref = client.create_memref_view_from_dlpack(dl)
+    dl = module.from_dlpack(memref)
+    print(f"-- Memref shape: {memref.shape}")
+    print(f"-- Memref dtype: {memref.dtype}")
+    print(f"-- {module.__name__}.from_dlpack(): {dl}")
+    return memref, dl
+
+
+print("Test keeping np.array, dlpack, memref view and dlpack view alive.")
+arr = np.array([1, 2, 3, 4], dtype=np.int32)
+dl = arr.__dlpack__()
+memref1, dl1 = create_memref_dlpackview_from_dlpack(dl, np)
```

Comment:
OK I think I understand... in Python we don't have control over the destruction order, so these could be destroyed prior to client. The test should be more explicit here then. Create a new client then explicitly delete it prior to deleting the memref object.

If this is indeed the issue, then we need a more robust fix to ensure the correct destruction order? I guess we somehow already side step this issue with non-dlpack memrefs

Reply:
@christopherbate @jhalakpatel  I have updated the PR issue description with the 2 issues that this PR solves. 
I can fix the test to add more explicit destruction order. We do not really need to correct the destruction order since we are only bookkeeping the external allocated `dlpack` tensors and if `RuntimeClient` is destructed early it shouldn't affect the behavior of these tensors.